types/validator_set.go

```diff
@@ -366,6 +383,266 @@ func (vals *ValidatorSet) Iterate(fn func(index int, val *Validator) bool) {
 	}
 }
 
+// If more or equal than 1/3 of total voting power changed in one block, then
+// a light client could never prove the transition externally. See
+// ./lite/doc.go for details on how a light client tracks validators.
+func (currentSet *ValidatorSet) UpdateWithChangeList(updates []*Validator) error {
```

Comment:
Will `UpdateWithChangeList` become obsolete?

Reply:
Not sure if we want to keep both APIs. Until we decide I leave it here.

Reply:
Suggestion: I would still remove it, when it is not used anywhere. If the need arises to treat updates as a list, we can readd it, or, open up a way for the application to control how updates are treated.